e2e/cli_test.go

```diff
@@ -0,0 +1,141 @@
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+
+	"gotest.tools/assert"
+	"gotest.tools/icmd"
+)
+
+const (
+	GoldenOutFilename = "output.regexp-golden"
+)
+
+func TestRunCLIScenarios(t *testing.T) {
+	// Find all the test fixtures with the expected files to run
+	err := filepath.Walk("./pkg/compose/tests/fixtures/", func(path string, info os.FileInfo, err error) error {
+
+		// Ignore directories
+		if info != nil && info.IsDir() {
+			return nil
+		}
+		filename := filepath.Base(path)
+		if filename == "docker-compose.yml" || filename == "docker-compose.yaml" {
+			// Use the parent dir as the testcase name
+			tcName := filepath.Base(filepath.Dir(path))
+			t.Run(tcName, func(t *testing.T) {
+				runScenario(t, path)
+			})
+
+		}
+		return nil
+	})
+	assert.NilError(t, err)
+}
+
+func runScenario(t *testing.T, composeFilename string) {
+	// Check for golden files
+	dir := filepath.Dir(composeFilename)
+	goldenFilename, err := filepath.Abs(filepath.Join(dir, GoldenOutFilename))
+	assert.NilError(t, err)
+	_, err = os.Stat(goldenFilename)
+	if err != nil && os.IsNotExist(err) {
+		t.Skipf("%s does not contain %s", dir, GoldenOutFilename)
+	}
+	var out strings.Builder
+	// Scenario:
+	stackname := filepath.Base(dir)
+	// * Run through a standard set of operations with a single aggregate stdout/stderr output - w/marker for each command
+	// * docker stack deploy $path $name
+	cmd := []string{"stack", "deploy", "-c", composeFilename, stackname}
+	result := icmd.RunCommand("docker", cmd...)
+	defer icmd.RunCommand("docker", "stack", "rm",
+		stackname)
+	result.Assert(t, icmd.Success)
+	out.WriteString("CMD: " + strings.Join(cmd, " ") + "\n")
+	out.WriteString(result.Combined())
+	// * Loop for $timeout until docker stack service matches golden
+	// TODO - need to support converging until it works
```

Comment:
My thinking with the markers is this can be broken into chunks so we can wait for convergence once the status  and resourcereporting parts are wired up on the back-end.